### PR TITLE
[fga] bump version to redo migrations

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Bumps the FGA version to initialize the migration again for everyone, so we can be sure the relationships are correct after updating spicedb.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 11ee265</samp>

Update and simplify the fine-grained access control (FGA) system for workspaces. Increase the `RelationshipUpdater` version and remove redundant code from `gitpod-server-impl.ts`.

</details>

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -445,23 +443,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 `operation not permitted: missing ${op} permission on ${resource.kind}`,
             );
         }
-        if (resource.kind === "workspace" && op === "get") {
```
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:448]: :tada:

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

File: components/server/src/authorization/relationship-updater.ts [hunk 2]
```diff
@@ -112,18 +111,7 @@ export class RelationshipUpdater {
     }
 
     private async isMigrated(user: User) {
-        const isMigrated = user.fgaRelationshipsVersion === RelationshipUpdater.version;
-        if (isMigrated) {
-            const hasSelfRelationship = await this.authorizer.find(rel.user(user.id).self.user(user.id));
-            if (!hasSelfRelationship) {
-                log.warn({ userId: user.id }, `User is marked as migrated but doesn't have self relationship.`);
```
Comment (svenefftinge) [inline components/server/src/authorization/relationship-updater.ts:119]: no entry in logs for the past 7 days. so we don't need this code path anymore

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 3]
```diff
@@ -445,23 +443,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
...
-            // access to workspaces is granted. Let's verify that this is also thecase with FGA
-            const result = await this.auth.hasPermissionOnWorkspace(
-                this.userID!,
-                "read_info",
-                resource.subject.id,
-                true,
-            );
-            if (!result) {
-                const isShared = await this.auth.find(rel.workspace(resource.subject.id).shared.anyUser);
-                log.error("user has access to workspace, but not through FGA", {
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:458]: We have a better check in authorizer that verifies the `shared` relationship exists after it was written.

**Pull request merged**